src/systems/kinetic_energy_monitor/KineticEnergyMonitor.cc

```diff
@@ -161,12 +171,54 @@ void KineticEnergyMonitor::Configure(const Entity &_entity,
     _ecm.CreateComponent(this->dataPtr->linkEntity,
         components::WorldAngularVelocity());
   }
+
+  // Create a halt motion component if one is not present.
+  if (!_ecm.Component<components::HaltMotion>(
+        _ecm.ParentEntity(this->dataPtr->linkEntity)))
+  {
+    _ecm.CreateComponent(_ecm.ParentEntity(this->dataPtr->linkEntity),
+        components::HaltMotion(false));
+  }
 }
 
 //////////////////////////////////////////////////
-void KineticEnergyMonitor::PostUpdate(const UpdateInfo &/*_info*/,
+void KineticEnergyMonitor::Update(const UpdateInfo &_info,
+                                 EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("KineticEnergyMonitor::Update");
+
+  // \TODO(anyone) Support rewind
+  if (_info.dt < std::chrono::steady_clock::duration::zero())
+  {
+    ignwarn << "Detected jump back in time ["
+        << std::chrono::duration_cast<std::chrono::seconds>(_info.dt).count()
+        << "s]. System may not work properly." << std::endl;
+  }
+
+  if (_info.paused)
+    return;
+
+  // Update Halt Motion component
+  auto *haltMotionComp =
+    _ecm.Component<components::HaltMotion>(
+      _ecm.ParentEntity(this->dataPtr->linkEntity));
+
+  if (this->dataPtr->haltMode &&
```

Comment:
Should we check if `haltMotionComp` is nullptr before dereferencing it?

Reply:
Moved to https://github.com/osrf/subt/pull/858 and fixed